### PR TITLE
[1LP][RFR] Adding dialog text area regex validation test

### PR DESCRIPTION
**Pull request by niyazRedhat**
### PRT Run
{{pytest: cfme/tests/services/test_dialog_regex_validation_in_catalog.py::test_dialog_element_regex_validation }}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 5464933142f1e4fe49ed630d68c1764ced01b2db

The *local* fixture **`dialog_textarea_cat_item`** is used in the following files:
- **cfme/tests/services/test_dialog_regex_validation_in_catalog.py**
    - *test_dialog_text_area_regex_validation*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/services/test_dialog_regex_validation_in_catalog.py [hunk 1]
```diff
@@ -55,20 +55,59 @@ def dialog_cat_item(appliance, catalog, request):
...
+        dialog=sd)
+    yield catalog_item, element_data, sd
+    if catalog_item.exists:
+        catalog_item.delete()
+    sd.delete_if_exists()
+
+
+@pytest.mark.parametrize("dialog_textarea_cat_item", [{"validation": "^([a-z0-9]+_*)*[a-z0-9]+$"}],
+                         indirect=True)
+def test_dialog_text_area_regex_validation(appliance, dialog_textarea_cat_item):
```
Comment (john-dupuy) [inline cfme/tests/services/test_dialog_regex_validation_in_catalog.py:101]: From what I can tell, the only difference between this test and `test_dialog_element_regex_validation` are the fixture parameters. Is there any way we can combine these test cases?

Comment (niyazRedhat) [reply cfme/tests/services/test_dialog_regex_validation_in_catalog.py:101]: I will give a try

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 59dcc185cb862025dbdb962b8488de5a5c0678ab

The *local* fixture **`dialog_cat_item`** is used in the following files:
- **cfme/tests/services/test_dialog_regex_validation_in_catalog.py**
    - *test_dialog_element_regex_validation*
    - *test_dialog_regex_validation_button*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/services/test_dialog_regex_validation_in_catalog.py [hunk 2]
```diff
@@ -25,16 +25,15 @@ def dialog_cat_item(appliance, catalog, request):
             'ele_label': fauxfactory.gen_alphanumeric(15, start="ele_label_"),
             'ele_name': fauxfactory.gen_alphanumeric(15, start="ele_name_"),
             'ele_desc': fauxfactory.gen_alphanumeric(15, start="ele_desc_"),
-            'choose_type': "Text Box"
+            "choose_type": hasattr(request, "param") and request.param.get
```
Comment (mshriver) [inline cfme/tests/services/test_dialog_regex_validation_in_catalog.py:28]: I don't think this is behaving quite in the way you intended.

the value for the dict field will be set to the boolean result of `hasattr() and request.param.get()` - not the string value of `request.param.get()`

**Comment by dajoRH** [comment]
I detected some fixture changes in commit a983062b154dad8b8b59afb91419f4e45e81e7a0

The *local* fixture **`dialog_cat_item`** is used in the following files:
- **cfme/tests/services/test_dialog_regex_validation_in_catalog.py**
    - *test_dialog_element_regex_validation*
    - *test_dialog_regex_validation_button*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/services/test_dialog_regex_validation_in_catalog.py [hunk 3]
```diff
@@ -16,30 +16,30 @@
 
 @pytest.fixture(scope="function")
 def dialog_cat_item(appliance, catalog, request):
+    default_data = {"choose_type": "Text Box", "validation": "^([a-z0-9]+_*)*[a-z0-9]+$"}
+    data = getattr(request, "param", default_data)
```
Comment (mshriver) [inline cfme/tests/services/test_dialog_regex_validation_in_catalog.py:20]: Nice fix, this pattern is much better.

**Pull request merged**